src/app/appRouting.module.ts

```diff
@@ -0,0 +1,50 @@
+import { NgModule } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+import { LoginComponent } from './pages/login/login.component';
+import { RegistrationComponent } from './pages/registration/registration.component';
+import { AppLayoutComponent } from './pages/app-layout/app-layout.component';
+import { CoursesComponent } from './pages/courses/courses.component';
+import { CourseEditComponent } from './pages/course-edit/course-edit.component';
+// import { CourseViewComponent } from './shared/components';
+import { CourseViewComponent } from './pages/course-view/course-view.component';
+import { AuthGuard } from './auth/guards/auth.guard';
+
+const routes: Routes = [
+  {
+    path: '',
+    component: AppLayoutComponent,
+    children: [
+      { path: '', redirectTo: '/courses', pathMatch: 'full' },
+      {
+        path: 'courses',
+        loadChildren: () =>
+          import('./pages/courses/courses.module').then((module) => module.CoursesModule),
+        canActivate: [AuthGuard],
```

Comment:
you can apply this guard on parent route so you don't have to apply it to every child route